app/cdap/components/WranglerGrid/SelectColumnPanel/DataTable/index.tsx

```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import { TableBody, Checkbox, Divider, Box } from '@material-ui/core';
+import React from 'react';
+import T from 'i18n-react';
+import TableRowWidget from 'components/WranglerGrid/SelectColumnPanel/DataTable/TableRow';
+import { IColumnTableProps } from 'components/WranglerGrid/SelectColumnPanel/DataTable/types';
+import {
+  ADD_TRANSFORMATION_PREFIX,
+  SELECT_COLUMN_LIST_PREFIX,
+} from 'components/WranglerGrid/SelectColumnPanel/constants';
+import { MULTI_SELECTION_COLUMN } from 'components/WranglerGrid/SelectColumnPanel/constants';
+import { TableContainer, Table, TableHead, TableRow, TableCell } from '@material-ui/core';
+import { grey } from '@material-ui/core/colors';
+import styled from 'styled-components';
+import { NormalFont, SubHeadBoldFont } from 'components/common/TypographyText';
+import { NoDataSVG } from 'components/GridTable/iconStore';
+
+const SelectColumnTableContainer = styled(TableContainer)`
+  height: 100%;
+`;
+
+const SelectColumnTable = styled(Table)`
+  display: flex;
+  flex-direction: column;
+`;
+
+const SelectColumnTableHead = styled(TableHead)`
+  height: 54px;
+`;
+
+const SelectColumnTableRow = styled(TableRow)`
+  font-weight: 400;
+  font-size: 16px;
+  line-height: 150%;
+  letter-spacing: 0.15px;
+  color: ${grey[700]};
+  display: grid;
+  grid-template-columns: 8% 45% 45%;
+  align-items: center;
+  height: 100%;
+`;
+
+const SelectColumnTableHeadCell = styled(TableCell)`
+  &.MuiTableCell-head {
+    padding: 0;
+    font-weight: 400;
+    font-size: 16px;
+    line-height: 150%;
+    letter-spacing: 0.15px;
+    color: ${grey[900]};
+    border-bottom: none !important;
+  }
+`;
+
+const FlexWrapper = styled(Box)`
+  display: flex;
+  height: 100%;
+  align-items: center;
+`;
+
+const CenterAlignBox = styled(Box)`
+  text-align: center;
+`;
+
+export default function({
+  columns,
+  transformationDataType,
+  onSingleSelection,
+  selectedColumns,
+  dataQualityValue,
+  isSingleSelection,
+  handleDisableCheckbox,
+  onMultipleSelection,
+  setSelectedColumns,
+  transformationName,
+}: IColumnTableProps) {
+  const handleChange = () => {
+    if (
+      MULTI_SELECTION_COLUMN?.filter(
+        (option) => option.value === transformationName && option.isMoreThanTwo === false
+      ).length > 0
+    ) {
+      if (selectedColumns.length) {
```

Comment:
this behavior is extremely confusing - ismorethanTwo is just saying whether its a radio button or a multi checkbox so I don't understand why you'd want to check the length of selected and then if its more than 2 return a slice of the first two elements? I have a feeling that a lot of this stuff is unnecessarily complicated and if it isn't then this needs a lot more comments so you don't have to dig through the code to understand whats going on.

Reply:
This function is used when we are having checkboxes to select column 
1. If we cannot select more than two column the first if is executed in which we check if column is already selected - if yes then we will deselect them and if not then first two column will be selected
2. in second condition we check if if more than two column can be selected - if already column selected than deselect them or select all column